File: frontend/js/tests/builders/DedoBuilder.js
function DedoBuilder() {
    this._direcao = [0.0, 0.0, 0.0];
    this._posicaoDaPonta = [0.0, 0.0, 0.0];
    this._velocidadeDaPonta = [0.0, 0.0, 0.0];
    this._apontando = false;
    this._tipo = DedoBuilder.TipoDedo.DEDAO;
}
DedoBuilder.TipoDedo = {
    DEDAO: 0,
    INDICADOR: 1,
    MEIO: 2,
    ANELAR: 3,
    MINDINHO: 4
};

DedoBuilder.prototype = {
    _direcao: undefined,
    _tipo: DedoBuilder.TipoDedo.DEDAO,
    _apontando: false, 
    _posicaoDaPonta: undefined,
    _velocidadeDaPonta: undefined,

    comDirecao: function(direcao) {
        this._direcao = direcao;
        return this;
    },

    doTipo: function(tipo) {
        this._tipo = tipo;
        return this;
    },

    construir: function() {
        return {
            direction: this._direcao,
            type: this._tipo
        }
    }
};

function _dedoDoTipo(tipo) {
    var direcao = Math.random() * tipo; // direcao random
    return new DedoBuilder()
        .comDirecao([direcao, direcao, direcao])
        .doTipo(tipo)
        .construir();
}

DedoBuilder.dedao = function() { return _dedoDoTipo(DedoBuilder.TipoDedo.DEDAO); };
DedoBuilder.indicador = function() { return _dedoDoTipo(DedoBuilder.TipoDedo.INDICADOR); };
DedoBuilder.meio = function() { return _dedoDoTipo(DedoBuilder.TipoDedo.MEIO); };
DedoBuilder.anelar = function() { return _dedoDoTipo(DedoBuilder.TipoDedo.ANELAR); };
DedoBuilder.mindinho = function() { return _dedoDoTipo(DedoBuilder.TipoDedo.MINDINHO); };

module.exports = DedoBuilder;

File: backend/Testes/Comum/Builders/Dominio/Sinais/Maos/DedoBuilder.cs
using Dominio.Sinais.Maos;

namespace Testes.Comum.Builders.Dominio.Sinais.Maos
{
    public class DedoBuilder
    {
        private double[] direcaoDaPonta;
        private double[] posicaoDaPonta;
        private TipoDedo tipo;

        public DedoBuilder ComDirecao(double[] direcaoDaPonta)
        {
            this.direcaoDaPonta = direcaoDaPonta;
            return this;
        }

        public DedoBuilder ComPosicaoDaPonta(double[] posicaoDaPonta)
        {
            this.posicaoDaPonta = posicaoDaPonta;
            return this;
        }

        public DedoBuilder DoTipo(TipoDedo tipo)
        {
            this.tipo = tipo;
            return this;
        }

        public Dedo Construir()
        {
            return new Dedo
            {
                Direcao = direcaoDaPonta,
                PosicaoDaPonta = posicaoDaPonta,
                Tipo = tipo
            };
        }

        public static Dedo Dedao()
        {
            return new DedoBuilder()
                .ComDirecao(DirecaoDaPonta())
                .DoTipo(TipoDedo.Dedao)
                .Construir();
        }

        public static Dedo Indicador()
        {
            return new DedoBuilder()
                .ComDirecao(DirecaoDaPonta())
                .DoTipo(TipoDedo.Indicador)
                .Construir();
        }

        public static Dedo Meio()
        {
            return new DedoBuilder()
                .ComDirecao(DirecaoDaPonta())
                .DoTipo(TipoDedo.Meio)
                .Construir();
        }

        public static Dedo Anelar()
        {
            return new DedoBuilder()
                .ComDirecao(DirecaoDaPonta())
                .DoTipo(TipoDedo.Anelar)
                .Construir();
        }

        public static Dedo Mindinho()
        {
            return new DedoBuilder()
                .ComDirecao(DirecaoDaPonta())
                .DoTipo(TipoDedo.Mindinho)
                .Construir();
        }

        private static double[] DirecaoDaPonta()
        {
            return new ArrayDeDouble().ComTamanho(3).Construir();
        }

        public static Dedo[] DedosPadroes()
        {
            return new[] 
            { 
                DedoBuilder.Dedao(), 
                DedoBuilder.Indicador(), 
                DedoBuilder.Meio(), 
                DedoBuilder.Anelar(), 
                DedoBuilder.Mindinho() 
            };
        }
    }
}
